deployment/src/main/java/io/quarkiverse/quinoa/deployment/PackageManager.java

```diff
@@ -127,15 +127,22 @@ public void run() {
     }
 
     public static PackageManager autoDetectPackageManager(Optional<String> binary, Path directory) {
-        String resolved = binary.orElse("npm");
-        if (Files.isRegularFile(directory.resolve("yarn.lock"))) {
-            resolved = "yarn";
-        }
-        if (Files.isRegularFile(directory.resolve("pnpm-lock.yaml"))) {
-            resolved = "pnpm";
+        String resolved = null;
+        if (binary.isEmpty()) {
+            if (Files.isRegularFile(directory.resolve("yarn.lock"))) {
+                resolved = "yarn";
+            } else if (Files.isRegularFile(directory.resolve("pnpm-lock.yaml"))) {
+                resolved = "pnpm";
+            } else {
+                resolved = "npm";
+            }
+            if (System.getProperty("os.name").startsWith("Windows")) {
```

Comment:
You may have a NPE here:

```suggestion
	    final String os = System.getProperty("os.name");
            if (os != null && os.startsWith("Windows")) {
```